postgres-appliance/major_upgrade/pg_upgrade.py

```diff
@@ -117,9 +117,12 @@ def update_extensions(self):
         for d in self._get_all_databases():
             conn_kwargs['dbname'] = d
             with get_connection_cursor(**conn_kwargs) as cur:
-                cur.execute('SELECT quote_ident(extname) FROM pg_catalog.pg_extension')
-                for extname in cur.fetchall():
-                    query = 'ALTER EXTENSION {0} UPDATE'.format(extname[0])
+                cur.execute('SELECT quote_ident(extname), extversion FROM pg_catalog.pg_extension')
+                for extname, version in cur.fetchall():
+                    # require manual update to 5.X+
+                    if extname == 'pg_partman' and int(version[0]) < 5:
+                        continue
```

Comment:
We could consider adding a log message here, instead of silently skipping the update. It might be helpful in debugging / informing users.
```suggestion
                        logger.info("Skipping update of '%s' v%s extension in database=%s", extname, version, d)
                        continue
```

Reply:
I would even make it a warning

Reply:
Agreed :+1: